ubi_manifest/worker/tasks/depsolver/rpm_depsolver.py

```diff
@@ -0,0 +1,150 @@
+import os
+import logging
+from concurrent.futures import as_completed
+from pubtools.pulplib import Criteria
+from more_executors.futures import f_proxy
+from more_executors import Executors
+from .pulp_queries import search_modulemds, search_rpms
+from .utils import create_or_criteria, get_n_latest_from_content, parse_bool_deps
+
+_LOG = logging.getLogger(__name__)
+
+# need to set significantly lower batches for general rpm search
+# otherwise db may very likely hit OOM error.
+BATCH_SIZE_RPM = int(os.getenv("UBI_MANIFEST_BATCH_SIZE_RPM", "15"))
+BATCH_SIZE_RESOLVER = int(os.getenv("UBI_MANIFEST_BATCH_SIZE_RESOLVER", "150"))
+
+
+class Depsolver:
+    def __init__(self, repos):
+
+        self.repos = repos
```

Comment:
Given that this is all py3-only, you could consider using type hints for all these.  In my experience it helps with IDE autocomplete.  Just thought I'd point out the option is there.